frontend/src/metabase/dashboard/components/GridLayout.jsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable react/prop-types */
+import React, { useCallback, useMemo } from "react";
+import ReactGridLayout from "react-grid-layout";
+import _ from "underscore";
+
+import { color } from "metabase/lib/colors";
+
+import "react-grid-layout/css/styles.css";
+import "react-resizable/css/styles.css";
+
+function GridLayout({
+  items,
+  itemRenderer,
+  layout,
+  cols,
+  width: gridWidth,
+  margin,
+  rowHeight,
+  isEditing,
+  ...props
+}) {
+  const cellSize = useMemo(() => {
+    const marginSlotsCount = cols - 1;
+    const totalHorizontalMargin = marginSlotsCount * margin;
+    const freeSpace = gridWidth - totalHorizontalMargin;
+    return {
+      width: freeSpace / cols,
+      height: rowHeight,
+    };
+  }, [cols, gridWidth, rowHeight, margin]);
+
+  const background = useMemo(() => {
```

Comment:
NIT: it could be a utils function outside of the component what would make it smaller

Reply:
Good point! Moved it to a new file